### ISSUE TITLE
Error in updateProgress for media player within search results

### ISSUE THREAD

**Issue by christianmemije** [learningequality/kolibri#2027]
## Summary

* Error in updateProgress for media player within search results 

## System information

 - Version: 0.5

## Traceback or relevant snippet from server.log or browser console

```
vue.runtime.esm.js:430 [Vue warn]: Error in beforeDestroy hook: "TypeError: Cannot read property 'id' of undefined"

found in

---> <VideoRender> at kolibri/plugins/media_player/assets/src/views/index.vue
       <ContentRender> at kolibri/core/assets/src/views/content-renderer/index.vue
         <LearnContent> at kolibri/plugins/learn/assets/src/views/content-page/index.vue
           <CoreBase> at kolibri/core/assets/src/views/core-base.vue
             <Learn> at kolibri/plugins/learn/assets/src/views/index.vue
               <Root>


vue.runtime.esm.js:519 TypeError: Cannot read property 'id' of undefined
    at contentNodeId (http://localhost:3000/js-dist/learn_module-0.6.dev020170824182606-git.js:9219:71)
    at Vue$3.Watcher.lazy (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:23657:16)
    at Watcher.get (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:5959:25)
    at Watcher.evaluate (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:6059:21)
    at VueComponent.computedGetter [as contentNodeId] (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:23661:19)
    at VueComponent.updateProgress (http://localhost:3000/js-dist/learn_module-0.6.dev020170824182606-git.js:9193:137)
    at boundFn (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:3349:14)
    at VueComponent.invoker (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:4918:18)
    at VueComponent.Vue.$emit (http://localhost:3000/js-dist/default_frontend-0.6.dev020170824182606-git.js:5387:16)
    at VueComponent.Vue.(anonymous function) [as $emit] (chrome-extension://nhdogjmejiglipccpnnnanhbledajbpd/build/backend.js:359:28)

```

## How to reproduce

1. Search for something within learn.
2. Click on video
3. Go back
4. See console

**Comment by rtibbles** [comment]
Guessing something is not getting torn down quickly enough, or is getting
torn down incompletely?

Not sure if this is a blocker for 0.5.x, as it is an edge case, unless the
error causes user noticeable issues?

**Comment by rtibbles** [comment]
Fixed in #2034
